packages/patterns-react/src/patterns/TableOfContents/TableOfContents.js

```diff
@@ -0,0 +1,271 @@
+   *
+   * @param {object} e event object
+   * @param {string} id data-driverlocation value of content
+   * @memberof TableOfContents
+   */
+  onMenuItemsClick = (e, id) => {
+    e.preventDefault();
+    e.stopPropagation();
+    document
+      .querySelector(`[data-driverlocation="${id}"]`)
+      .scrollIntoView({ behavior: 'smooth', block: 'start' });
+  };
+
+  /**
+   * handle onChange event on select to scroll to element
+   *
+   * @param {object} e event object
+   * @memberof TableOfContents
+   */
+  onSelectChange = e => {
+    this.setState({ selectedItem: e.target.value });
+    const child = document.querySelector(
+      `[data-driverlocation="${e.target.value}"]`
+    );
+    child.scrollIntoView(true);
+    const scrolledY = window.scrollY;
+    if (scrolledY) {
+      window.scroll({
+        left: 0,
+        top: scrolledY - 48,
+        behavior: 'smooth',
+      });
+    }
+  };
+
+  /**
+   * Set focus state to menuRef to activate
+   *
+   * @memberof TableOfContents
+   */
+  onMenuButtonClick = () => {
+    this.menuRef.current.focus();
+  };
+
+  /**
+   * Render options for select
+   *
+   * @param {Array} items menu item arrray
+   * @memberof TableOfContents
+   * @returns {*} JSX Object
+   */
+  renderSelectOptions = items => {
+    return items.map(item => {
+      if (item) {
+        return (
+          <option
+            className={`${prefix}--tableofcontents__mobile-menu__select__option`}
+            key={item.id}
+            value={item.id}>
+            {item.title}
+          </option>
+        );
+      }
+    });
+  };
+
+  /**
+   * Render items for menu
+   *
+   * @param {Array} items menu item arrray
+   * @returns {*} JSX Object
+   */
+  renderMenuItems = items => {
+    return items.map(item => {
+      if (item) {
+        return (
+          <li
+            key={item.id}
+            className={classNames(
+              `${prefix}--tableofcontents__desktop-menu__item`,
+              this.setActiveClass(this.state.selectedItem, item.id)
+            )}>
+            <a
+              onClick={e => {
+                this.onMenuItemsClick(e, item.id);
+              }}
+              href={item.id}>
+              {item.title}
+            </a>
+          </li>
+        );
+      }
+    });
+  };
+
+  /**
+   * Render TableOdContents pattern
+   *
+   * @returns {*} JSX Object
+   * @memberof TableOfContents
+   */
+  render() {
+    const {
+      layoutType,
+      menuItems,
+      children,
+      marginTop,
+      marginBottom,
+    } = this.props;
+
+    return featureFlag(
+      TABLE_OF_CONTENTS,
+      <section
+        data-autoid={`${stablePrefix}--tableofcontents`}
+        className={`${prefix}--tableofcontents`}>
+        <div className={`${prefix}--tableofcontents__mobile-menu`}>
+          <div className={`${prefix}--tableofcontents__mobile-menu__wrapper`}>
+            <span className={`${prefix}--tableofcontents__mobile-menu__label`}>
+              Jump to{' '}
```

Comment:
`Jump to` shouldn't be hardcoded english. We should set as a prop, and can default to `Jump to`.

Reply:
Yeah, We will still need to update it.